test/suites/dev-tanssi/registrar/test_registrar_proxy.ts

```diff
@@ -0,0 +1,210 @@
+                    return a.event.method == "ProxyAdded";
+                });
+                expect(ev1.length).to.be.equal(1);
+
+                const proxies = await polkadotJs.query.proxy.proxies(bob.address);
+                expect(proxies.toJSON()[0]).to.deep.equal([
+                    {
+                        delegate: charlie.address,
+                        proxyType: "Registrar",
+                        delay: 0,
+                    },
+                ]);
+
+                // Use proxy
+                await context.createBlock();
+                const currentSesssion = await polkadotJs.query.session.currentIndex();
+                const sessionDelay = await polkadotJs.consts.registrar.sessionDelay;
+                const expectedScheduledOnboarding =
+                    BigInt(currentSesssion.toString()) + BigInt(sessionDelay.toString());
+
+                const emptyGenesisData = () => {
+                    const g = polkadotJs.createType("TpContainerChainGenesisDataContainerChainGenesisData", {
+                        storage: [
+                            {
+                                key: "0x636f6465",
+                                value: "0x010203040506",
+                            },
+                        ],
+                        name: "0x436f6e7461696e657220436861696e2032303030",
+                        id: "0x636f6e7461696e65722d636861696e2d32303030",
+                        forkId: null,
+                        extensions: "0x",
+                        properties: {
+                            tokenMetadata: {
+                                tokenSymbol: "0x61626364",
+                                ss58Format: 42,
+                                tokenDecimals: 12,
+                            },
+                            isEthereum: false,
+                        },
+                    });
+                    return g;
+                };
+                const containerChainGenesisData = emptyGenesisData();
+
+                // assert we can inject on chain data with proxy
+                const tx2 = polkadotJs.tx.proxy.proxy(
+                    bob.address,
+                    null,
+                    polkadotJs.tx.registrar.register(2002, containerChainGenesisData)
+                );
+                await context.createBlock([await tx2.signAsync(charlie)]);
+                // Check that the on chain genesis data is set correctly
+                const onChainGenesisData = await polkadotJs.query.registrar.paraGenesisData(2002);
+                // TODO: fix once we have types
+                expect(emptyGenesisData().toJSON()).to.deep.equal(onChainGenesisData.toJSON());
+
+                // assert we can inject bootnodes with proxy
+                const tx3 = polkadotJs.tx.proxy.proxy(
+                    bob.address,
+                    null,
+                    polkadotJs.tx.registrar.setBootNodes(2002, ["dummy"])
```

Comment:
Sorry, in #341 we moved the setBootNodes to a different pallet, pallet_data_preservers. I guess we want that to be included in the SudoRegistrar proxy as well. So this should work:

```suggestion
                    polkadotJs.tx.dataPreservers.setBootNodes(2002, ["dummy"])
```

Reply:
yes exactly